subsys/net/buf.c

```diff
@@ -161,11 +160,11 @@ const struct net_buf_data_cb net_buf_fixed_cb = {
 
 #if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
 
-static u8_t *heap_data_alloc(struct net_buf *buf, size_t size, s32_t timeout)
+static u8_t *heap_data_alloc(struct net_buf *buf, size_t *size, s32_t timeout)
 {
 	u8_t *ref_count;
 
-	ref_count = k_malloc(1 + size);
+	ref_count = k_malloc(1 + *size);
 	if (!ref_count) {
 		return NULL;
 	}
```

Comment:
Should assign to *size?

Reply:
No, since this size is what gets set to buf->size, i.e. it should not take into account internal metadata such as the refcount.